frontend/server/src/Controllers/Tag.php

```diff
@@ -65,4 +65,42 @@ public static function getPublicTags() {
             'problemTag'
         );
     }
+
+    public static function getFrequentTagsByLevel(
+        string $problemLevel
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::TAGS_LIST,
+            'frequent',
+            fn () => \OmegaUp\DAO\Tags::getFrequentsByLevel(
+                $problemLevel
+            ),
+            APC_USER_CACHE_SESSION_TIMEOUT
+        );
+    }
+
+    /**
+     * Return most frequent public tags of a certain level
+     *
+     * @return list<array{name: string}>
+     */
+    public static function apiFrequentTags(\OmegaUp\Request $r): array {
+        $param = '';
+        if (is_string($r['level'])) {
+            $param = $r['level'];
+        } else {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterEmpty',
+                'level'
+            );
+        }
```

Comment:
Todo eso se puede sustituir por:
```suggestion
        $param = $r->ensureString('level')
```
Esa función ya se encarga de arrojar la excepción en caso de no encontrar nada o que no sea una cadena válida